### ISSUE TITLE
[Bug]: Bad json parse

### ISSUE THREAD

**Issue by bdecarne** [microsoft/graphrag#1173]
### Do you need to file an issue?

- [X] I have searched the existing issues and this bug is not already filed.
- [ ] My model is hosted on OpenAI or Azure. If not, please look at the "model providers" issue and don't file a new one here.
- [X] I believe this is a legitimate bug, not just a question. If this is a question, please use the Discussions area.

### Describe the bug

When receive response from LLM (ollama + gemma2), the function try_parse_json_object parse it wrong due to weak regex 

Response:

```
```json
{
    "points": [
        {"description": "This report compares the attractiveness of several French cities for business and their quality of life. It specifically mentions Lille, Rennes, Lorient, and Braga (Portugal).", "score": 100},
        {"description": "According to the data, Lille is considered more attractive for business than Rennes [Data: Reports (0)].", "score": 75},
        {"description": "Both Rennes and Braga are highly ranked for quality of life [Data: Reports (0)] .", "score": 75},
        {"description": "Lorient ranks higher than Rennes in an unspecified quality of life ranking [Data: Reports (0)].", "score": 50}
    ]
}
```

Snippet from `try_parse_json_object`  (`llm/openai/utils.py`) : 

```python
_pattern = r"\{(.*)\}"
_match = re.search(_pattern, input)
input = "{" + _match.group(1) + "}" if _match else input

```

Result :

```json
{"description": "This report compares the attractiveness of several French cities for business and their quality of life. It specifically mentions Lille, Rennes, Lorient, and Braga (Portugal).", "score": 100}
```

Expected :

```json
{
    "points": [
        {"description": "This report compares the attractiveness of several French cities for business and their quality of life. It specifically mentions Lille, Rennes, Lorient, and Braga (Portugal).", "score": 100},
        {"description": "According to the data, Lille is considered more attractive for business than Rennes [Data: Reports (0)].", "score": 75},
        {"description": "Both Rennes and Braga are highly ranked for quality of life [Data: Reports (0)] .", "score": 75},
        {"description": "Lorient ranks higher than Rennes in an unspecified quality of life ranking [Data: Reports (0)].", "score": 50}
    ]
}
```

### GraphRAG Config Used

```yaml
# Paste your config here

```

### Additional Information

- GraphRAG Version:
- Operating System:
- Python Version:
- Related Issues:

**Comment by AlonsoGuevara** [comment]
Thanks for the detailed report! Indeed, I'll proceed to fix this!

**Comment by AlonsoGuevara** [comment]
Fixed on #1218
